src/platform/web/Platform.js

```diff
@@ -82,6 +82,31 @@ async function loadOlmWorker(paths) {
     return olmWorker;
 }
 
+// needed for mobile Safari which shifts the layout viewport up without resizing it
+// when the keyboard shows (see https://bugs.webkit.org/show_bug.cgi?id=141832)
+function adaptUIOnVisualViewportResize(container) {
+    if (!window.visualViewport) {
+        return;
+    }
+    window.visualViewport.addEventListener('resize', () => {
+        const scrollable = container.querySelector('.bottom-aligned-scroll');
+        let scrollTopBefore, heightBefore, heightAfter;
+
+        if (scrollable) {
+            scrollTopBefore = scrollable.scrollTop;
+            heightBefore = scrollable.getBoundingClientRect().height;
```

Comment:
```suggestion
            heightBefore = scrollable.offsetHeight;
```
Is a bit more concise and _might_ be faster as the whole bounding box doesn't need to be calculated. Same below.